chat-server/server.js

```diff
@@ -21,8 +23,68 @@ const welcomeMessage = {
 //We will start with one message in the array.
 const messages = [welcomeMessage];
 
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.post("/messages", (req, res) => {
+  const newMessage = req.body;
+  if (
+    !newMessage.hasOwnProperty("text") ||
+    !newMessage.hasOwnProperty("from") ||
+    newMessage.from === "" ||
+    newMessage.text === "" ||
+    typeof newMessage.text != "string" ||
+    typeof newMessage.from != "string"
+  ) {
+    res.sendStatus(400).send("wrong input");
```

Comment:
One suggestion for improvement is that instead of using `res.sendStatus(400).send("wrong input")`, you can use `res.status(400).send("wrong input")`. `res.status(400)` set the status code and then send the error message using `send`.

Reply:
Ahh I see, that makes more sense. Thank you.